lib/Util.ts

```diff
@@ -3,27 +3,27 @@ import * as path from 'path';
 
 /**
  * creates an url from a request url and optional base url (http://server:8080)
- * @param {string} requestUrl - a fully qualified url or relative url
+ * @param {string} resource - a fully qualified url or relative path
  * @param {string} baseUrl - an optional baseUrl (http://server:8080)
  * @return {string} - resultant url 
  */
-export function getUrl(requestUrl: string, baseUrl?: string): string  {
+export function getUrl(resource: string, baseUrl?: string): string  {
     if (!baseUrl) {
-        return requestUrl;
+        return resource;
     }
 
     let base: url.Url = url.parse(baseUrl);
 
-    // requestUrl (specific per request) always wins
-    let combined: url.Url = url.parse(requestUrl);
-    combined.protocol = combined.protocol || base.protocol;
-    combined.auth = combined.auth || base.auth;
-    combined.host = combined.host || base.host;
+    // resource (specific per request) eliments take priority
+    let resultantUrl: url.Url = url.parse(resource);
+    resultantUrl.protocol = resultantUrl.protocol || base.protocol;
+    resultantUrl.auth = resultantUrl.auth || base.auth;
+    resultantUrl.host = resultantUrl.host || base.host;
 
-    combined.pathname = path.join(base.pathname, combined.pathname);
+    let basePathComponent: string = base.pathname === '/' ? '': base.pathname;
+    resultantUrl.pathname = path.posix.join(basePathComponent, resultantUrl.pathname);
```

Comment:
use resolve.  drive letter on windows not a problem.  If you do something illogical like using a drive letter on windows, you will get a failure which is expected.  We're optimizing for the positive rooted and unrooted cases.

Reply:
I see what it's doing now.